ta4j-core/src/main/java/org/ta4j/core/indicators/AroonOscillatorIndicator.java

```diff
@@ -35,15 +35,20 @@
  */
 public class AroonOscillatorIndicator extends CachedIndicator<Num> {
 
+    public static final String KEYWORD = "AroonOscillator";
```

Comment:
Such things will increase the memory consumption on every instance creation beforehand. I am against this. If the user needs a string representation, then it should be withint `toString()`.

Reply:
I will remove anything not directly related to this PR.

I don't think there is any difference between putting the string in a static field or hard-coding it in the toString().  The string "AroonOscillator" will exist once in the JVM one way or the other; I suspect all JVMs intern strings.   Caching the value to be returned by toString() is a common approach when performance matters.  It likely doesn't in this case.

As for increasing memory consumption...  consider this... the indicators use each other as children so often, duplicating array lists and calculations, that very little I will do will be important...  I will remove it anyway.  cheers

Reply:
@johnthetireless Thanks. Btw, you can use **JOL** to measure object sizes or create a heap dump and analyze by "eclipse memory analyzer"(https://www.baeldung.com/jvm-measuring-object-sizes).

Reply:
Thanks.  I have looked at the memory analyzer briefly but I haven't used it for an actual dump yet.
My little "representation" field would have added the size of a reference/pointer to the indicator object: 8 bytes, probably.  The string itself would be interned somewhere by the JVM it seems...

		public static void main(String[] args) {
			String a = "AroonOscillator";
			String b = "AroonOscillator";
			System.out.println(a.equals(b));  	// true, of course
			System.out.println(a == b);		// also true; my JVM interns strings			
		}